platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/EconomyHook.java

```diff
@@ -0,0 +1,331 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.plugin.bukkit.hooks;
+
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.util.Locale;
+import java.util.Optional;
+import me.lokka30.treasury.api.common.service.Service;
+import me.lokka30.treasury.api.common.service.ServiceRegistry;
+import me.lokka30.treasury.api.economy.EconomyProvider;
+import me.lokka30.treasury.api.economy.account.PlayerAccount;
+import me.lokka30.treasury.api.economy.currency.Currency;
+import me.lokka30.treasury.api.economy.response.EconomySubscriber;
+import me.lokka30.treasury.plugin.bukkit.TreasuryBukkit;
+import org.bukkit.OfflinePlayer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class EconomyHook implements TreasuryPAPIHook {
+
+    private EconomyProvider provider;
+    private final DecimalFormat format = new DecimalFormat("#,###");
+    private final PAPIExpansion expansion;
+    private final TreasuryBukkit plugin;
+    private final String k, m, b, t, q;
+    private Baltop baltop;
+
+    public EconomyHook(PAPIExpansion expansion, TreasuryBukkit plugin) {
+        this.expansion = expansion;
+        this.plugin = plugin;
+        this.k = expansion.getString("formatting.thousands", "k");
+        this.m = expansion.getString("formatting.millions", "m");
+        this.b = expansion.getString("formatting.billions", "b");
+        this.t = expansion.getString("formatting.trillions", "t");
+        this.q = expansion.getString("formatting.quadrillions", "q");
```

Comment:
aren't M, B, T and Q all meant to be capitalized?

Reply:
They are, and it makes it even more clear that this is a modified copy of PAPI's Vault expansion. All the file headers should contain GPL notice.
https://github.com/PlaceholderAPI/Vault-Expansion/blob/63ddf694fdc1c99940d49e513469cb5780cfba24/src/main/java/com/extendedclip/papi/expansion/vault/VaultExpansion.java#L58-L62
https://github.com/PlaceholderAPI/Vault-Expansion/blob/63ddf694fdc1c99940d49e513469cb5780cfba24/src/main/java/com/extendedclip/papi/expansion/vault/VaultEcoHook.java#L73-L77

Reply:
This is not a crucial issue, since the APIs remains intact. However, I am not pleased that we need to borrow code, especially that from a more restrictive license. I think this is something important we need to address before merging.

Reply:
It's used in the plugin side that API users won't depend on, it should be fine imo. It does get a bit messy, but if we're using the same placeholder design we probably shouldn't bother reinventing the wheel.